File: easy/28_Find_the_Index_of_the_First_Occurrence_in_a_String.c
int ft_strlen(char *str){
    int i = 0;
    while (str[i])
        i++;
    return i;
}
int strStr(char *haystack, char *needle) {
    int j = 0;
    int lh = ft_strlen(haystack);
    int ln = ft_strlen(needle);

    if (ln <= lh) {
        while (j < lh) {
            int i = 0;
            if (haystack[j] == needle[i]) {
                int k = j;
                while (i < ln) {
                    if (haystack[k] == needle[i]) {
                        i++;
                        if (k < lh - 1) {
                            k++;
                        }
                    } else {
                        break;
                    }
                }
                if (i == ln) {
                    return j;
                }
            }
            j++;
        }
    }
    return -1;
}

File: easy/28_Find_the_Index_of_the_First_Occurrence_in_a_String.py
def strStr(haystack: str, needle: str) -> int:
    j = 0
    lh = len(haystack)
    ln = len(needle)
    if ln <= lh:
        while j < lh:
            i = 0
            if haystack[j] == needle[i]:
                k = j
                while i < ln:
                    if haystack[k] == needle[i]:
                        i += 1
                        if k < lh - 1:
                            k += 1
                    else:
                        break
                if i == ln:
                    return j
            j += 1
    return -1
